internal/resources/ibm/en_destination_huawei.go

```diff
@@ -0,0 +1,109 @@
+package ibm
+
+import (
+	"fmt"
+
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+// EnDestinationHuawei struct
+//
+// Resource information: https://cloud.ibm.com/catalog/services/event-notifications#about
+// Pricing information: https://cloud.ibm.com/catalog/services/event-notifications
+type EnDestinationHuawei struct {
+	Address   string
+	IsPreProd bool
+	Name      string
+	Plan      string
+	Region    string
+}
+
+// EnDestinationHuaweiUsageSchema defines a list which represents the usage schema of EnDestinationHuawei.
+var EnDestinationHuaweiUsageSchema = []*schema.UsageItem{}
+
+// PopulateUsage parses the u schema.UsageData into the EnDestinationHuawei.
+// It uses the `infracost_usage` struct tags to populate data into the EnDestinationHuawei.
+func (r *EnDestinationHuawei) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+// BuildResource builds a schema.Resource from a valid EnDestinationHuawei struct.
+// This method is called after the resource is initialised by an IaC provider.
+// See providers folder for more information.
+func (r *EnDestinationHuawei) BuildResource() *schema.Resource {
+	costComponents := []*schema.CostComponent{
+		EnDestinationHuaweiPushDestinationInstancesCostComponent(r),
+	}
+
+	return &schema.Resource{
+		Name:           r.Address,
+		UsageSchema:    EnDestinationHuaweiUsageSchema,
+		CostComponents: costComponents,
+	}
+}
+
+func EnDestinationHuaweiPushDestinationInstancesCostComponent(r *EnDestinationHuawei) *schema.CostComponent {
+
+	var costComponent schema.CostComponent
+	component_name := "Push Destination Instances"
+	unit := "PUSH_DESTINATION_INSTANCES"
```

Comment:
it looks like the firefox, huwawei, and chrome, android, iphone, and safari all use the same price and metric (PUSH_DESTINATION_INSTANCES)? it's a lot of redundant code. Would be nice if the code could be reduced for them all to share. not a blocker, but it's a lot of files getting added

Reply:
Yeah, I definitely agree... I wasn't sure how to consolidate this, since they are different resources and so they use different schemas.